aspnet/3-binary-data/Controllers/BooksController.cs

```diff
@@ -127,11 +127,21 @@ public ActionResult Edit(long? id)
         // POST: Books/Edit/5
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public ActionResult Edit(Book book, long id)
+        public async Task<ActionResult> Edit(Book book, long id, HttpPostedFileBase image)
         {
             if (ModelState.IsValid)
             {
                 book.Id = id;
+                if (image != null)
+                {
+                    var imageUrl = await _imageUploader.UploadImage(image, book.Id);
+                    book.ImageUrl = imageUrl;
+                }
+                else
+                {
+                    // Avoid losing the image.  Keep the original image.

```

Comment:
Good catch!  Hmm ...

I've been researching and I don't see an established pattern for MVC with model bindings with images and retaining the file path on Edit.

I've been playing around with ways to persist this.  To make this `Edit()` action simple for demonstration, what are your thoughts on this?

**BooksController.cs**
```cs
if (image != null)
{
  // ...
}
_store.Update(book);
```

***Form.cshtml***
```cshtml
<div class="form-group">
    @Html.Label("Cover Image")
    <input type="file" id="image" name="image" class="form-control">
    @Html.HiddenFor(model => model.ImageUrl)
</div>

// Book.cs
[Bind(Include = "Title, Author, PublishedDate, Description, ImageUrl")]
class Book
```

Thoughts?

Else, if we want to do it in the controller (as is) then :+1:

This might make the line a little bit more clear / simple:

```cs
book.ImageUrl = _store.Read(book.Id).ImageUrl;

// book.Id for clarity
// ? null conditional not required because this book must exist to view the Edit form
```